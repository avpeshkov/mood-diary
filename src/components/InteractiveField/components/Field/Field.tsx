import React, { FC } from "react";
import styled from "@emotion/styled";
import type { FieldProps } from "types/field";

import { Cell } from "./components";

const FieldWrapper = styled.div`
    display: inline-block;
    padding: 10px;
    border: 2px solid lightgray;
`;

export const Field: FC<FieldProps> = ({ field, onClick }) => (
    <FieldWrapper>
        {field.map((row, y) => [
            ...row.map((filled: string, x) => (
                <Cell key={`${x}_${y}`} x={x} y={y} onClick={onClick}>
                    {filled}
                </Cell>
            )),
            y !== row.length - 1 ? <br key={y} /> : null,
        ])}
    </FieldWrapper>
);
