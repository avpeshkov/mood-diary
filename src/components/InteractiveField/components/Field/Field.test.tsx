import React from "react";
import { mount } from "enzyme";
import renderer from "react-test-renderer";

import { Field } from "./Field";

describe("Field", () => {
    it("renders cells for passed empty field", () => {
        expect(
            renderer
                .create(
                    <Field
                        field={[
                            ["", ""],
                            ["", ""],
                        ]}
                        onClick={jest.fn()}
                    />
                )
                .toJSON()
        ).toMatchSnapshot();
    });
    it("renders filled cells snapshot check", () => {
        expect(
            renderer
                .create(
                    <Field
                        field={[
                            ["x", "o"],
                            ["o", ""],
                        ]}
                        onClick={jest.fn()}
                    />
                )
                .toJSON()
        ).toMatchSnapshot();
    });
    it("renders filled cells", () => {
        const field = mount(
            <Field
                field={[
                    ["x", "o"],
                    ["o", ""],
                ]}
                onClick={jest.fn()}
            />
        );
        expect(field.findWhere((el) => el.html() === "x" && typeof el.type() !== "string").length).toBe(1);
        expect(field.findWhere((el) => el.html() === "o" && typeof el.type() !== "string").length).toBe(2);
    });
    it("passed onClick inside cells", () => {
        const onClick = jest.fn();
        const field = mount(<Field field={[["", "x", "o"]]} onClick={onClick} />);
        field.find("button:empty").simulate("click");
        expect(onClick).toHaveBeenCalledWith(0, 0);
    });
});
