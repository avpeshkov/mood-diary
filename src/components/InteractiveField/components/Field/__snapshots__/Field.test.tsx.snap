// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Field renders cells for passed empty field 1`] = `
.emotion-4 {
  display: inline-block;
  padding: 10px;
  border: 2px solid lightgray;
}

.emotion-0 {
  width: 25px;
  height: 25px;
  border: 1px solid;
  display: inline-block;
  border-radius: 10px;
  line-height: 25px;
  text-align: center;
  margin: 5px;
  vertical-align: bottom;
  border-color: gray;
}

<div
  className="emotion-4"
>
  <button
    className="emotion-0"
    onClick={[Function]}
  >
    
  </button>
  <button
    className="emotion-0"
    onClick={[Function]}
  >
    
  </button>
  <br />
  <button
    className="emotion-0"
    onClick={[Function]}
  >
    
  </button>
  <button
    className="emotion-0"
    onClick={[Function]}
  >
    
  </button>
</div>
`;

exports[`Field renders filled cells snapshot check 1`] = `
.emotion-4 {
  display: inline-block;
  padding: 10px;
  border: 2px solid lightgray;
}

.emotion-3 {
  width: 25px;
  height: 25px;
  border: 1px solid;
  display: inline-block;
  border-radius: 10px;
  line-height: 25px;
  text-align: center;
  margin: 5px;
  vertical-align: bottom;
  border-color: gray;
}

.emotion-0 {
  width: 25px;
  height: 25px;
  border: 1px solid;
  display: inline-block;
  border-radius: 10px;
  line-height: 25px;
  text-align: center;
  margin: 5px;
  vertical-align: bottom;
  border-color: lightgray;
  color: gray;
}

<div
  className="emotion-4"
>
  <button
    className="emotion-0"
    onClick={[Function]}
  >
    x
  </button>
  <button
    className="emotion-0"
    onClick={[Function]}
  >
    o
  </button>
  <br />
  <button
    className="emotion-0"
    onClick={[Function]}
  >
    o
  </button>
  <button
    className="emotion-3"
    onClick={[Function]}
  >
    
  </button>
</div>
`;
